app/models/partner.rb

```diff
@@ -360,4 +362,11 @@ def opening_times_is_json_or_nil
 
     errors.add :base, 'Partner.opening_times must be valid json'
   end
+
+  def three_or_less_category_tags
+    number_of_category_tags = tags.count { |t| t.type == 'Category' }
```

Comment:
this should just be `categories.count` if STI is set up properly!

Reply:
I'm sorry I don't understand this, searching STI has not helped either...

Reply:
Single Table Inheritance (in case anyone else is looking)